backend.go

```diff
@@ -23,11 +24,77 @@ type jsonFileBackend struct {
 	filename string
 }
 
+type flagJson struct {
```

Comment:
This is cute, it's like a merge of both old and new flags!

It might be better to use `Version` to select old vs new? But I'm uncertain, happy to keep this. We should add a comment to explain what this is for, since it's not obvious.

Reply:
i thought about versioning but if its ok with you, i think this just might be simpler for now! i added a comment explaining what was happening

Reply:
Yup, that's fine